spec/controllers/registrations_controller_spec.rb

```diff
@@ -120,6 +120,96 @@
       end
     end
   end
+
+  describe '#update' do
+    let!(:user) { FactoryGirl.create(:user) }
+
+    context 'valid update' do
+      before(:each) do
+        user.confirm
+        sign_in user
+        patch :update,
+          params: { user: { email: 'new@test.com', password: '123456', password_confirmation: '123456', current_password: user.password } },
+          xhr: true
+        user.reload
+      end
+
+      it 'responds HTTP 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'update email' do
+        expect(user.email).to eq('new@test.com')
+      end
+
+      it 'update password' do
+         expect(user.valid_password?('123456')).to be(true)
+      end
+    end
+
+    context 'invalid update' do
+      before(:each) do
+        user.confirm
+        sign_in user
+      end
+
+      context 'incorrect current password' do
+        before(:each) do
+          patch :update,
+            params: { user: { email: 'new@test.com', password: '123456', password_confirmation: '123456', current_password: '111111' } },
+            xhr: true
+          user.reload
+        end
+
+        it 'responds HTTP 200' do
+          expect(response).to have_http_status(200)
```

Comment:
If your request fails you shouldn't expect a success 200 code. You should receive a fail http code `422 : unprocessable_entity `

Reply:
I need respond with http 200 because the responds has format js and the errors are added to the html dynamical